lib/money/currency.rb

```diff
@@ -34,18 +35,17 @@ def initialize(method, currency, attribute)
     class UnknownCurrency < ArgumentError; end
 
     class << self
-      alias_method :original_new, :new
       def new(id)
         id = id.to_s.downcase
         unless stringified_keys.include?(id)
           raise UnknownCurrency, "Unknown currency '#{id}'"
         end
 
-        @@mutex.synchronize { instances[id] } || super
+        _instances[id] || @@mutex.synchronize { _instances[id] ||= super }
       end
 
-      def instances
-        @instances ||= Hash.new { |h, k| h[k] = original_new(k) }
+      def _instances
```

Comment:
It's just an accessor for easier debugging/development/tests, that's all. That's also why I prefixed it with `_`.